# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/ipc/WebrtcTCPSocketParent.h
# Commit: da41f0fab427
# Full Hash: da41f0fab427b199f1a0c6156467a0ad9a9e6891
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Make proxy config parameters optional in WebrtcTCPSocket to get the NAT simulator working again. r=mjf,mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45104
# ==============================================================================

diff -r b83124cf3a89 -r da41f0fab427 media/mtransport/ipc/WebrtcTCPSocketParent.h
--- a/media/mtransport/ipc/WebrtcTCPSocketParent.h	Tue Sep 17 18:17:24 2019 +0000
+++ b/media/mtransport/ipc/WebrtcTCPSocketParent.h	Tue Sep 17 18:17:37 2019 +0000
@@ -26,8 +26,8 @@
   mozilla::ipc::IPCResult RecvAsyncOpen(
       const nsCString& aHost, const int& aPort, const nsCString& aLocalAddress,
       const int& aLocalPort, const bool& aUseTls,
-      const LoadInfoArgs& aLoadInfoArgs, const nsCString& aAlpn,
-      const int& aProxyPolicy) override;
+      const Maybe<LoadInfoArgs>& aLoadInfoArgs, const Maybe<nsCString>& aAlpn,
+      const Maybe<int>& aProxyPolicy) override;
 
   mozilla::ipc::IPCResult RecvWrite(nsTArray<uint8_t>&& aWriteData) override;
 
@@ -35,7 +35,7 @@
 
   void ActorDestroy(ActorDestroyReason aWhy) override;
 
-  explicit WebrtcTCPSocketParent(dom::TabId aTabId);
+  explicit WebrtcTCPSocketParent(const Maybe<dom::TabId>& aTabId);
 
   // WebrtcTCPSocketCallback
   void OnClose(nsresult aReason) override;